### PR TITLE
Fix make check

### PR DESCRIPTION
**Pull request by zbeekman**
`make check` fails for silly reasons. This fixes the makefile in tests/serial.

### CODE REVIEW

**Comment by sfilippone** [comment]
No, the correct fix is to change BASEDIR into INSTALLDIR. 
Rationale: the INSTALLDIR variable was copied from another subdirectory in tests, and I forgot to doublecheck; on my system it worked anyway because it was picking up a previous installation. 
The idea is that when building, it points to the build tree, but when (some of) the test subdirs are copied onto INSTALLDIR/Samples, they then point to the installed modules, not to the build directory. I have updated the Makefile accordingly, and have adjusted the tag. 
Thanks for the heads up.

**Comment by zbeekman** [comment]
Hi Salvatore,

Whatever the correct fix is, your `make check` is still broken. The issue is that the run directory logic is broken too: in a new build it doesn't exist.

Also, just FYI, best practice for yanked releases is to mark them yanked and bump the patch version, just so people aren't confused.

File: test/serial/Makefile [hunk 1]
```diff
@@ -12,21 +12,22 @@ LDLIBS=$(PSBLDLIBS)
 FINCLUDES=$(FMFLAG)$(MODDIR) $(FMFLAG).
 
 
-EXEDIR=./runs
+EXEDIR=./runs/
```
Comment (zbeekman) [inline test/serial/Makefile]: This needs changing in your fix, b/c if the directory doesn't exist d_matgen gets copied to an executable named `runs`

File: test/serial/Makefile [hunk 2]
```diff
@@ -12,21 +12,22 @@ LDLIBS=$(PSBLDLIBS)
...
 
 
 all: d_matgen 
 
 psb_d_xyz_impl.o d_matgen.o: psb_d_xyz_mat_mod.o 
 
-d_matgen: d_matgen.o psb_d_xyz_mat_mod.o psb_d_xyz_impl.o   
+d_matgen: d_matgen.o psb_d_xyz_mat_mod.o psb_d_xyz_impl.o
+	mkdir $(EXEDIR) || true
```
Comment (zbeekman) [inline test/serial/Makefile]: EXEDIR is not created during install, nor does it exist in the distributed tarball or source tree. It needs to be created if it's missing

File: test/serial/Makefile [hunk 3]
```diff
@@ -12,21 +12,22 @@ LDLIBS=$(PSBLDLIBS)
...
 	$(FLINK) $(LINKOPT) d_matgen.o psb_d_xyz_mat_mod.o psb_d_xyz_impl.o \
 	 -o d_matgen  $(PSBLAS_LIB) $(LDLIBS) 
 	/bin/cp -p $(CPUPDFLAG) d_matgen $(EXEDIR)	
 #	/bin/mv d_matgen $(EXEDIR)	
 
 check: all
-	cd runs && echo 5 | ./d_matgen
+	cd $(EXEDIR) && echo 5 | ./d_matgen
```
Comment (zbeekman) [inline test/serial/Makefile]: what's the point of the EXEDIR variable if you don't use it?

**Comment by sfilippone** [comment]
In other subdirs  the ./runs subdir is there with input files, so it does
not need to be created anew.
Can you fix it and bump the patch number? I have to rush away, will  be
fully online tomorrow.

Thanks
Salvatore

**Pull request closed without merging**